Python/ceval.c

```diff
@@ -6976,22 +6976,23 @@ PyObject *
 PyEval_GetGlobals(void)
 {
     PyThreadState *tstate = _PyThreadState_GET();
-    _PyInterpreterFrame *current_frame = tstate->cframe->current_frame;
-    if (current_frame == NULL) {
+    _Py_framedata *fdata = tstate->cframe->current_frame;
+    if (fdata == NULL) {
+        _PyErr_SetString(tstate, PyExc_SystemError, "cannot get globals; frame data does not exist");
```

Comment:
Ah, this was the origin of the PyImport_ImportModule failure - in main, failing to get the locals sets an error, but failing to get the globals doesn't.